content/blog/2023/2023-03-31-serialize-deserialize-with-jsonview.md

```diff
@@ -0,0 +1,617 @@
+
+````java
+public class User {
+    @JsonView(Views.GetView.class)
+    private String loginName;
+
+    @JsonView(Views.GetView.class)
+    private String firstName;
+
+    @JsonView(Views.GetView.class)
+    private String lastName;
+
+    @JsonView(Views.PatchView.class)
+    private String mobileNo;
+}
+
+public class Views {
+    
+    public static interface PatchView {
+        
+    }
+
+    public static interface GetView extends PatchView {
+
+    }
+}
+````
+For the above views to apply to the HTTP methods, we will define our controller methods as below:
+
+````java
+@RestController
+public class UserController {
+
+   @PostMapping(path = "/userdetails")
+   public ResponseEntity<User> post(
+           @RequestBody @JsonView(value = Views.GetView.class) User user) {
+   return ResponseEntity.status(HttpStatus.CREATED).body(savedUser);
+}
+
+
+   @PatchMapping(path = "/userdetails/{userId}")
+   public ResponseEntity<?> patch(
+           @RequestBody @JsonView(value = Views.PatchView.class) User user) {
+   return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
+ }
+
+}
+````
+This configuration will allow only the **@JsonView mapped fields** to be updated in the POST and PATCH requests respectively. Other fields will be ignored.
+
+In the further sections, we will look at a sample Spring Boot application to understand how to use `@JsonView` in the context of the use cases defined above.
+
+## What is Serialization and Deserialization
+- **Serialization** is the process of converting an object into a stream of bytes.
+- **Deserialization** is the process of converting the serialized form of an object back to a copy of the original object.
+
+## Serialize and Deserialize objects in a Spring Boot Application
+Serialization and deserialization form the core of REST APIs. 
+**Spring Boot internally uses Jackson's `ObjectMapper` class to perform serialization and deserialization.**
+
+Deserialization example:
+{{% image alt="settings" src="images/posts/spring-boot-jsonview/Deserialise.JPG" %}}
+
+Serialization example:
+{{% image alt="settings" src="images/posts/spring-boot-jsonview/Serialise.JPG" %}}
+
+## Spring Boot defaults for @JsonView configuration
+The `ObjectMapper` class uses **`MapperFeature.DEFAULT_VIEW_INCLUSION`** to determine how the `JsonView` annotation needs to behave.
+**This configuration will determine whether the properties that are not annotated with @JsonView should be included during serialization and deserialization.**
+
+For the sample application, we will use the Spring Boot version 2.7.5 that internally uses [Jackson 2.13](https://fasterxml.github.io/jackson-databind/javadoc/2.13/com/fasterxml/jackson/databind/MapperFeature.html).
+**By default, the MapperFeature.DEFAULT_VIEW_INCLUSION is set to true which means when enabling a JSON View, non-annotated fields also get serialized.**
+{{% image alt="settings" src="images/posts/spring-boot-jsonview/MapperFeature.JPG" %}}
```

Comment:
The picture of the source code is unnecessary. The text above provides enough information.